javaslang/src/main/java/javaslang/collection/Vector.java

```diff
@@ -694,10 +694,10 @@ public int indexOf(T element, int from) {
 
     @Override
     public Vector<T> init() {
-        if (isEmpty()) {
-            throw new UnsupportedOperationException("init of empty Vector");
-        } else {
+        if (nonEmpty()) {
```

Comment:
I don't think we should try to predict these things, the JIT gods work in mysterious ways, it's the reason I'm always measuring.
I think we can consider inlining to be for free in these cases, it's a simple jump, it's certainly not the bottleneck :)